retrieval/discovery/marathon/marathon_test.go

```diff
@@ -22,23 +22,22 @@ import (
 	"golang.org/x/net/context"
 
 	"github.com/prometheus/prometheus/config"
-	"github.com/prometheus/prometheus/retrieval/discovery/marathon"
 )
 
 var marathonValidLabel = map[string]string{"prometheus": "yes"}
 
-func newTestDiscovery(client marathon.AppListClient) (chan []*config.TargetGroup, *MarathonDiscovery) {
+func newTestDiscovery(client AppListClient) (chan []*config.TargetGroup, *Discovery) {
 	ch := make(chan []*config.TargetGroup)
-	md := NewMarathonDiscovery(&config.MarathonSDConfig{
+	md := &Discovery{
 		Servers: []string{"http://localhost:8080"},
-	})
-	md.client = client
+	}
+	md.Client = client
```

Comment:
Copied from how it was.
If I start fixing this, I'll give the entire thing a pass over – that's better off in a subsequent PR.